gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/role/impl/RoleRepositoryImpl.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.gateway.handler.common.role.impl;
+
+import io.gravitee.am.gateway.handler.common.role.RoleFacade;
+import io.gravitee.am.model.Role;
+import io.gravitee.am.service.RoleService;
+import io.reactivex.rxjava3.core.Single;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+import java.util.Set;
+
+public class RoleRepositoryImpl implements RoleFacade {
+
```

Comment:
I wouldn't mess artifact/word Repository with it.
How about RoleFacade interface and two impl: DegradededModeRoleFacade and RegularModeRoleFacade?

Reply:
why not naming it DefaultRoleManager and rename the one using the sync process as  InMemoryRoleManager?